src/Storages/MergeTree/GinIndexStore.cpp

```diff
@@ -81,14 +82,22 @@ UInt64 GinIndexPostingsBuilder::serialize(WriteBuffer & buffer)
     {
         rowid_bitmap.runOptimize();
         auto size = rowid_bitmap.getSizeInBytes();
+        auto buf = std::make_unique<char[]>(size);
+        rowid_bitmap.write(buf.get());
+
+        auto codec = CompressionCodecFactory::instance().get(GIN_COMPRESSION_CODEC, GIN_COMPRESSION_LEVEL);
+        Memory<> memory;
+        memory.resize(codec->getCompressedReserveSize(static_cast<UInt32>(size)));
```

Comment:
Is there any chance that `size` grows > `UINT32_MAX`?

Reply:
I don't think so, the posting lists size should be far less than maximum size of UInt32 since postings lists are built in segments which have limited sizes.